Sources/ScipioKit/Producer/FrameworkProducer.swift

```diff
@@ -304,6 +302,26 @@ struct FrameworkProducer {
         }
     }
 
+    private func buildTargets(
+        _ targets: OrderedSet<CacheSystem.CacheTarget>
+    ) async -> (OrderedSet<CacheSystem.CacheTarget>, (any Error)?) {
```

Comment:
Could you make a result model instead of returning tuple?

```swift
enum TargetBuildResult {
    case interrupted(OrderedSet<CacheSystem.CacheTarget>, any Error)
    case completed(OrderedSet<CacheSystem.CacheTarget>)
}
```